src/ide/editor/theme.ts

```diff
@@ -5,7 +5,48 @@
 // creating editor instances.
 //
 
-export let themes = {
+import { ThemeName } from "../tgui";
+
+export interface ThemeDefinition {
+	bars: {
+		linenumbers: {
+			background: string;
+			color: string;
+		};
+		icons: {
+			background: string;
+			color: string;
+		};
+		separator: string;
+	};
+	content: {
+		background: string;
+		highlight: [
+			string,
+			string,
+			string,
+			string,
+			string,
+			string,
+			string,
+			string,
+			string,
+			string
+		];
```

Comment:
I unfortunately didn't find any reason why this array has this length, so I kept it as a tuple of 10 strings.

Reply:
That's a rather quick and dirty design. The ten strings hold CSS color values like "#000". They should probably be named rather than accessed by index. Index 0 is "normal" style (no highlighting), index 1 is a thee keyword highlight color, ..., index 9 is a comment. If you feel like refactoring the collection then please go ahead.